netbox_cmdb/netbox_cmdb/models/interface.py

```diff
@@ -0,0 +1,140 @@
+
+    name = models.CharField(max_length=100)
+    device = models.ForeignKey(
+        to="dcim.Device",
+        on_delete=models.CASCADE,
+        related_name="%(class)sdevice",
+        null=False,
+        blank=False,
+    )
+    autonegotiation = models.BooleanField(default=True)
+    speed = models.PositiveIntegerField(
+        validators=[MinValueValidator(1), MaxValueValidator(800000000)],
+        blank=True,
+        null=True,
+        help_text="Interface speed in kB/s",
+    )
+    fec = models.CharField(
+        choices=FEC_CHOICES,
+        max_length=5,
+        blank=True,
+        null=True,
+    )
+    description = models.CharField(max_length=100, blank=True, null=True)
+
+    def __str__(self):
+        return f"{self.device.name}--{self.name}"
+
+    class Meta:
+        unique_together = ("device", "name")
+
+
+class LogicalInterface(ChangeLoggedModel):
+    """A logical interface configuration."""
+
+    index = models.PositiveSmallIntegerField()
+    parent_interface = models.ForeignKey(
+        to="DeviceInterface",
+        related_name="%(class)s",
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+    )
+    mtu = models.PositiveIntegerField(blank=True, null=True)
+    type = models.CharField(
+        choices=LOGICAL_INTERFACE_TYPE_CHOICES,
+        max_length=2,
+        default=None,
+    )
+    vrf = models.ForeignKey(
+        to="VRF", related_name="%(class)s_vrf", on_delete=models.CASCADE, blank=True, null=True
+    )
+    ipv4_address = models.ForeignKey(
+        to="ipam.IPAddress",
+        related_name="%(class)s_ipv4_address",
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+    )
+    ipv6_address = models.ForeignKey(
+        to="ipam.IPAddress",
+        related_name="%(class)s_ipv6_address",
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+    )
+    mode = models.CharField(
+        choices=LOGICAL_INTERFACE_MODE_CHOICES,
+        blank=True,
+        null=True,
+        default=None,
+        max_length=20,
+        help_text="Interface mode (802.1Q)",
+    )
+    untagged_vlan = models.ForeignKey(
+        to="VLAN",
+        related_name="%(class)s_untagged_vlan",
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+        default=None,
+    )
+    tagged_vlans = models.ManyToManyField(
+        to="VLAN", related_name="%(class)s_tagged_vlans", blank=True, default=None
+    )
+    native_vlan = models.ForeignKey(
+        to="VLAN",
+        related_name="%(class)s_native_vlan",
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+        default=None,
+    )
+    description = models.CharField(max_length=100, blank=True, null=True)
+
+    def __str__(self):
+        return f"{self.parent_interface.name}--{self.index}"
+
+    class Meta:
+        unique_together = ("index", "parent_interface")
+
+
+class PortLayout(ChangeLoggedModel):
```

Comment:
a docstring explaining the fields would be nice here.
The different names would not be obvious for someone who was not in the design phase.